tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -113,6 +112,15 @@ public void AddSpan(Span span)
                         var samplingDecision = Tracer.Sampler?.MakeSamplingDecision(span) ?? SamplingDecision.Default;
                         SetSamplingPriority(samplingDecision);
                     }
+
+                    if (span.TraceId128.Upper > 0)
+                    {
+                        // this is a 128-bit trace id, so add the "_dd.p.tid" ((P)ropagated (T)race (ID)) trace tag.
+                        // NOTE: we can't use the optimized HexString.ToHexString(span.TraceId.Upper)
+                        // because this tag CANNOT be zero-padded.
+                        var upperTraceIdHex = span.TraceId128.Upper.ToString("x");
+                        Tags.SetTag(Datadog.Trace.Tags.Propagated.TraceIdUpper, upperTraceIdHex);
```

Comment:
It's been a while so I just want to confirm, setting this propagated tag on this span will update the TraceTagsCollection (and emit it the context injection), correct?

Reply:
This line is not setting a tag on a span. `this.Tags` is the `TraceTagsCollection` :)

For context, this is not used for horizontal propagation (to downstream service), it's used for vertical propagation (to the trace agent). The serialized span schema only has a field for a 64-bit trace id, so we're sending the upper 64 bits as a new propagated tag, `_dd.p.tid`.